File: 1. Fundamental-Level/CSharp/03. OOP-November-2015/[HW]/04. Inheritance-And-Abstraction/04. [HW]/01. Human-Student-Worker/Student.cs
using System;

public class Student : Human
{
    private string facultyNumber;

    public Student(string firstName, string lastName, string facultyNumber)
        : base(firstName, lastName)
    {
        this.FacultyNumber = facultyNumber;
    }

    public string FacultyNumber
    {
        get
        {
            return this.facultyNumber;
        }
        set
        {
            if (value.Length < 4 ||
                value.Length > 10)
            {
                throw new ArgumentOutOfRangeException("The length of the faculty number must be in range [5..10].");
            }

            this.facultyNumber = value;
        }
    }

    public override string ToString()
    {
        return string.Format("{0} {1} - {2}", this.FirstName, this.LastName, this.facultyNumber);
    }
}

File: 1. Fundamental-Level/Java/OOP-Bacis-June-2016/[EX]/05. Inheritance/src/problem3_Mankind/Student.java
package problem3_Mankind;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Student extends Human {
    private String facultyNumber;

    public Student(String firstName, String lastName, String facultyNumber) {
        super(firstName, lastName);
        this.setFacultyNumber(facultyNumber);
    }

    private void setFacultyNumber(String facultyNumber) {
        if (!isValidFacultyNumber(facultyNumber)){
            throw new IllegalArgumentException("Invalid faculty number!");
        }

        this.facultyNumber = facultyNumber;
    }

    private boolean isValidFacultyNumber(String facultyNumber) {
        String regex = "^[\\da-zA-Z]{5,10}$";
        Pattern pattern = Pattern.compile(regex);
        Matcher matcher = pattern.matcher(facultyNumber);

        return matcher.find();
    }

    @Override
    public String toString() {
        return super.toString() +
                "Faculty number: " + this.facultyNumber + System.lineSeparator();
    }
}